examples/kalman_filter.ipynb

```diff
@@ -0,0 +1,490 @@
+    "    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For the sake of simplicity, here we assume that the dynamical system takes the form\n",
+    "\n",
+    "$\\frac{dx}{dt} = Ax(t) + Bu(t)$\n",
+    "\n",
+    "$y(t) = Cx(t) + \\epsilon$ \n",
+    "\n",
+    "where $A,B,C$ are constant matrices, i.e. the system is linear in its state $x(t)$ and its input $u(t)$. Further, the dynamics- and measurement-functions do not depend on time.\n",
+    "\n",
+    "Hence, the above represents the general form of (physical) *linear, time-invariant systems* (LTI systems)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here we define a container object for LTI systems."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class LTISystem(eqx.Module):\n",
+    "    A: jnp.ndarray\n",
+    "    B: jnp.ndarray\n",
+    "    C: jnp.ndarray\n",
+    "\n",
+    "    def __iter__(self):\n",
+    "        return (x for x in (self.A, self.B, self.C))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "An harmonic oscillator is an LTI system, this function returns such an LTI system."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def harmonic_oscillator(damping: float = 0.0, time_scaling: float = 1.0) -> LTISystem:\n",
+    "    A = jnp.array([[0.0, time_scaling], [-time_scaling, -2 * damping]])\n",
+    "    B = jnp.array([[0.0], [1.0]])\n",
+    "    C = jnp.array([[0.0, 1.0]])\n",
+    "    return LTISystem(A, B, C)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here we define some utility functions that allow us to simulate LTI systems."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "WARNING:absl:No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)\n"
+     ]
+    }
+   ],
+   "source": [
+    "def _interpolate_us(ts, us, B):\n",
+    "    if us:\n",
+    "        u_t = dfx.LinearInterpolation(ts=ts, ys=us)\n",
+    "    else:\n",
+    "        m = B.shape[-1]\n",
+    "        u_t = SimpleNamespace(evaluate=lambda t: jnp.zeros((m,)))\n",
+    "    return u_t\n",
+    "\n",
+    "\n",
+    "def _diffeqsolve(\n",
```

Comment:
Let's remove the underscore at the start of the name; I'd prefer simplicity instead of declaring public vs private here.